src/ray/gcs/gcs_server/gcs_server_main.cc

```diff
@@ -38,17 +41,40 @@ DEFINE_string(session_name,
               "session_name: The session name (ClusterID) of the cluster.");
 DEFINE_string(ray_commit, "", "The commit hash of Ray.");
 
+namespace {
+// GCS server output filename.
+constexpr std::string_view kGcsServerLog = "gcs_server.out";
+}  // namespace
+
 int main(int argc, char *argv[]) {
+  gflags::ParseCommandLineFlags(&argc, &argv, true);
+
+  // Backward compatibility notes:
+  // Due to historical reason, GCS server flushes all logging and stdout/stderr to a
+  // single file called `gcs_server.out`. To keep backward compatibility at best effort,
+  // we use the same filename as output, and disable log rotation by default.
+  const int64_t log_rotation_max_size = FLAGS_log_rotation_size <= 0
+                                            ? std::numeric_limits<int64_t>::max()
+                                            : FLAGS_log_rotation_size;
+  RAY_CHECK_EQ(setenv(
+      "RAY_ROTATION_MAX_BYTES", std::to_string(log_rotation_max_size), /*overwrite=*/1));
```

Comment:
can you explain diff of RAY_ROTATION_MAX_BYTES vs FLAGS_log_rotation_size ? If we already have the former, then we only need to fix existing behavior? I see gcs_server_main.cc already call ray::RayLog::StartRayLog and why does the log rotations in it do not work?

Reply:
If we don't do anything in this PR, instead only do 2 things:

1. remove python stdout/stderr redirection
2. change ray_log_shutdown_raii from `/*log_dir=*/""` to `/*log_dir=*/FLAGS_log_dir`

will the rotations automatically work?

Reply:
Yes but the file name will be changed. We want to keep the existing gcs_server.out filename for backward compatibility.

Reply:
> will the rotations automatically work?

To answer your question, passing the log directory works for log rotation.
But one motivation would be backward compatibility, namely keep the `gcs_server.out` filename.